### PR TITLE
Remove cosign pkg, take a dep on it.

### PR DESCRIPTION
**Pull request by vaikas**
Signed-off-by: Ville Aikas <vaikas@chainguard.dev>

#### Summary
Remove the pkg/ directories that have to do with verification, aka, cosign bits. Take a dependency
on it rather than even try to maintain two parallel implementations. Then as time passes, some of
these dependencies will make their way to sigstore/sigstore and some won't, but at least it's very
clear now where we take deps to cosign.

#### Ticket Link

Fixes

#### Release Note

```release-note

```

### CODE REVIEW

**Review by imjasonh — APPROVED** [review]
This looks great! It looks like there aren't any remaining deps on `cmd/cosign` at least, which is great. We should put in a test to ensure no new ones sneak in. 👍

File: pkg/webhook/validation.go [hunk 1]
```diff
@@ -27,11 +27,11 @@ import (
 	"knative.dev/pkg/apis"
 	"knative.dev/pkg/logging"
 
-	"github.com/sigstore/policy-controller/cmd/cosign/cli/fulcio/fulcioroots"
+	"github.com/sigstore/cosign/cmd/cosign/cli/fulcio/fulcioroots"
```
Comment (imjasonh) [inline pkg/webhook/validation.go:30]: This seems to be the only cmd/cosign dependency, and it's moving to sigstore/sigstore in https://github.com/sigstore/sigstore/pull/435 🎉

Comment (vaikas) [reply pkg/webhook/validation.go:30]: Yeah, can't wait!

**Comment by vaikas** [comment]
Once we get rid of the fulcioroots, I'll add that test, ok?

**Review by cpanato — APPROVED** [review]
amazing

**Pull request merged**